src/pkg/caendr/caendr/models/sql/strain.py

```diff
@@ -41,14 +44,71 @@ def __repr__(self):
 
   @staticmethod
   def to_sortable_strain(strain):
-    import re
-    m = re.match('([A-Z]+)([0-9]+)', strain.strain)
+    m = re.match(STRAIN_NAME_REGEX, strain.strain)
     if m:
-      g = m.groups()
-      return (g[0], int(g[1]))
-    else:
-      return ('', 0)
+      return (m.group(1), int(m.group(2)))
+    return ('', 0)
 
   @staticmethod
   def sort_by_strain(arr):
     return sorted(arr, key=Strain.to_sortable_strain)
+
+
+  @classmethod
+  def get_columns_ordered(cls, names_only=False):
+    '''
+      Get the list of columns in this table, in the order specified in the source data sheet.
+    '''
+
+    # Define the column order using the column names
+    col_order = [
+      'species',
+      'species_id_method',
+      'strain',
+      'isotype',
+      'previous_names',
+      'release',
+      'source_lab',
+      'latitude',
+      'longitude',
+      'landscape',
+      'locality_description',
+      'substrate',
+      'substrate_comments',
+      'substrate_temp',
+      'ambient_temp',
+      'ambient_humidity',
+      'associated_organism',
+      'inbreeding_state',
+      'sampled_by',
+      'isolated_by',
+      'sampling_date',
+      'sampling_date_comments',
+      'notes',
+      'strain_set',              # "set" in source data sheet
+      'issues',
+      'issue_notes',
+      'isotype_ref_strain',
+      'sequenced',               # "wgs_seq" in source data sheet
+      # 'peel-zeel',
+      # 'pha-sup',
+      # 'nagoya',
+    ]
+
+    # Get the list of columns, associated with their index in the ordered list
+    col_list = [
+      (col_order.index(c.name) if c.name in col_order else None, c)
+        for c in list(cls.__mapper__.columns)
+    ]
+
+    # Filter out any columns not in the ordered list
+    col_list = [ c for c in col_list if c[0] is not None]
+
+    # Sort by index in ordered list, and map back to just the column object
+    col_list = [ c[1] for c in sorted(col_list, key=lambda x: x[0]) ]
+
+    # Return names or full columns, as requested
+    # Can't just return ordered list above because some of the columns listed are not defined in this table
+    if names_only:
```

Comment:
prefer returning consistent data shapes (col vs col.name) -- eg: in this method, only return cols. Add a new method that returns the col names (could call this one first, etc).

Reply:
Fixed in https://github.com/AndersenLab/CAENDR/pull/373/commits/714c44ede5b1593590bded8331dd39f1741d4a76